fix: replace useBinder with useForm (#2756)
vaadin/docs: articles/lit/guides/forms/form-status-tracking.adoc

The following properties are available both for the form as a whole, and for each field independently.
They are a part of the <<reference#binder-node,`BinderNode`>> interface.
ifdef::react[]

> Assume you have the [classname]`UseFormResult` instance by calling [methodname]`useForm`, then you can access form status properties as:

Reviewer: ```suggestion
You can access form status properties for the whole form from the [classname]`UseFormResult` instance returned from [methodname]`useForm`:
```
Thanks for adding the distinction for React BTW 👍

[source,tsx]
----
const form = useForm(EntityModel);

const invalid = form.invalid;
const dirty = form.dirty;
const errors = form.errors;
...
----

> and the same goes for a single field when the [classname]`UseFormResult` instance obtained by calling [methodname]`useForm`, then you can access these properties as:

Reviewer: ```suggestion
To access form status properties for individual fields, use the [classname]`UseFormPartResult` instance returned from [methodname]`useFormPart`:
```

[source,tsx]
----
const form = useForm(EntityModel);

> const nameField = useFormPart(form.mode.name);

Reviewer: ```suggestion
const nameField = useFormPart(form.model.name);
```
